herbstluftwm: 0.7.2 -> 0.8.1

ThibautMarty opened the pull request:
###### Motivation for this change

Update herbstluftwm.
`CMAKE_INSTALL_SYSCONF_PREFIX` needs to be explicitly set for bash completion.
I put the documentation and man pages in their respective outputs.

###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [x] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [x] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [x] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

ThibautMarty committed:
herbstluftwm: 0.7.2 -> 0.8.0

Build now relies on cmake.
Added outputs "doc" and "man".
Setting CMAKE_INSTALL_SYSCONF_PREFIX is needed for bash completion
directory, otherwise it tries to put files in /etc/bash_completion.d

ThibautMarty commented:
As a draft because I did not have time to test it for now.

ThibautMarty commented:
CC maintainer, @the-kenny

cole-h commented on pkgs/applications/window-managers/herbstluftwm/default.nix:
```diff
@@ -1,23 +1,22 @@
-      --replace "/usr/share" "\$(PREFIX)/share"
+  outputs = [ "out" "doc" "man" ];
+
+  preConfigure = ''
```
I think it's more idiomatic to do the following:

```
cmakeFlags = [
    "-DCMAKE_INSTALL_SYSCONF_PREFIX=$out/etc"
];
```

(Make sure it actually does what it should, though; I could be wrong.)

ThibautMarty committed:
herbstluftwm: fix cross compilation

Doc generation is disabled by default when cross compiling because asciidoc
does not cross compile for now

ThibautMarty committed:
herbstluftwm: add option to the X Window System manual chapter

ThibautMarty commented:
Rebased onto master to avoid conflict with 84cf00f98031e93f389f1eb93c4a7374a33cc0a9

cole-h reviewed (approved):
Diff LGTM, builds fine.

```
[7 built, 19 copied (114.8 MiB), 48.2 MiB DL]
https://github.com/NixOS/nixpkgs/pull/84853
1 package blacklisted:
tests.nixos-functions.nixos-test

1 package built:
herbstluftwm
```

ThibautMarty committed:
herbstluftwm: 0.7.2 -> 0.8.1

Bash completions is now installed in $out/share/bash-completion.
Setting CMAKE_INSTALL_SYSCONF_PREFIX is still needed for
/etc/xdg/herbstluftwm installation.

ThibautMarty commented:
Updated to 0.8.1.
`CMAKE_INSTALL_SYSCONF_PREFIX` still needs to be set for `/etc/xdg/herbstluftwm/` installation.

vlaci commented on pkgs/applications/window-managers/herbstluftwm/default.nix:
```diff
@@ -1,27 +1,47 @@
-  nativeBuildInputs = [ pkgconfig ];
-  buildInputs = [ glib libX11 libXext libXinerama ];
+  buildInputs = [
+    glib
```
glib is no longer needed since the code has been rewritten to C++

Mic92 committed:
herbstluftwm: remove glib

no longer required, since it's C++ now.
